diff -urN 0ad-0.0.17-alpha/libraries/source/fcollada/src/Makefile 0ad-0.0.17-alpha-patched/libraries/source/fcollada/src/Makefile
--- 0ad-0.0.17-alpha/libraries/source/fcollada/src/Makefile	2014-09-25 01:39:28.000000000 +1100
+++ 0ad-0.0.17-alpha-patched/libraries/source/fcollada/src/Makefile	2014-10-14 20:01:50.380309151 +1100
@@ -9,9 +9,7 @@
 
 CXX ?= g++
 CXXFLAGS += -fvisibility=hidden -W -Wall -Wno-unused-parameter -Wno-unused-function $(OS_DEFINE) $(PIC_FLAGS) $(CPPFLAGS)
-CXXFLAGS_DEBUG := -O0 -g -D_DEBUG -DRETAIL
 CXXFLAGS_RELEASE := -O2 -DNDEBUG -DRETAIL
-CXXFLAGS_TEST := -O0 -g -D_DEBUG
 LIBS += `pkg-config libxml-2.0 --libs`
 INCLUDES += -IFCollada `pkg-config libxml-2.0 --cflags`
 INCLUDES_TEST := -IFCollada/FColladaTest $(INCLUDES)
@@ -219,33 +217,19 @@
 	FCollada/FColladaTest/FCTestXRef/FCTestXRefSimple.cpp \
 	FCollada/FColladaTest/FCTestXRef/FCTestXRefTree.cpp \
 
-OBJECTS_DEBUG = $(addprefix output/debug/,$(SOURCE:.cpp=.o))
 OBJECTS_RELEASE = $(addprefix output/release/,$(SOURCE:.cpp=.o))
-OBJECTS_TEST = $(addprefix output/test/,$(SOURCE:.cpp=.o) $(TEST_SOURCE:.cpp=.o))
-OBJECTS_ALL = $(OBJECTS_DEBUG) $(OBJECTS_RELEASE) $(OBJECTS_TEST)
+OBJECTS_ALL = $(OBJECTS_RELEASE)
 
-all: output/libFColladaSD.a output/libFColladaSR.a install
+all: output/libFColladaSR.a install
 
 output_dirs:
-	bash -c 'mkdir -p output/{debug,release,test}/{FCollada/{FCDocument,FMath,FUtils,FColladaTest/{FCTestAssetManagement,FCTestExportImport,FCTestXRef}},FColladaPlugins/FArchiveXML}'
-
-test: FCollada/FColladaTest/ output/FColladaTest
-	( cd FCollada/FColladaTest/ ; ../../output/FColladaTest )
-	cat FCollada/FColladaTest/FColladaTestLog.txt
-
-output/libFColladaSD.a: $(OBJECTS_DEBUG) | output_dirs
-	@echo "$@"
-	@ar -cr $@ $(OBJECTS_DEBUG); ranlib $@
+	bash -c 'mkdir -p output/release/{FCollada/{FCDocument,FMath,FUtils,FColladaTest/{FCTestAssetManagement,FCTestExportImport,FCTestXRef}},FColladaPlugins/FArchiveXML}'
 
 output/libFColladaSR.a: $(OBJECTS_RELEASE) | output_dirs
 	@echo "$@"
 	@ar -cr $@ $(OBJECTS_RELEASE); ranlib $@
 
-output/FColladaTest: $(OBJECTS_TEST) | output_dirs
-	$(CXX) -o $@ $(LDFLAGS) $(OBJECTS_TEST) $(LIBS)
-
-install: output/libFColladaSD.a output/libFColladaSR.a
-	cp output/libFColladaSD.a ../lib/libFColladaSD.a
+install: output/libFColladaSR.a
 	cp output/libFColladaSR.a ../lib/libFColladaSR.a
 
 dfile = $(@:.o=.d)
@@ -255,18 +239,10 @@
 		sed -e 's/\#.*//' -e 's/^[^:]*: *//' -e 's/ *\\$$//' \
 		    -e '/^$$/ d' -e 's/$$/ :/' < $(dfile) >> $(Pfile) && \
 		rm -f $(dfile)
-output/debug/%.o: %.cpp | output_dirs
-	@echo "$<"
-	$(CXX) $(CXXFLAGS) $(CXXFLAGS_DEBUG) $(INCLUDES) -MD -MF $(dfile) -c $< -o $@
-	$(gendep)
 output/release/%.o: %.cpp | output_dirs
 	@echo "$<"
 	$(CXX) $(CXXFLAGS) $(CXXFLAGS_RELEASE) $(INCLUDES) -MD -MF $(dfile) -c $< -o $@
 	$(gendep)
-output/test/%.o: %.cpp | output_dirs
-	@echo "$<"
-	$(CXX) $(CXXFLAGS) $(CXXFLAGS_TEST) $(INCLUDES_TEST) -MD -MF $(dfile) -c $< -o $@
-	$(gendep)
 
 clean:
 	rm -rf output
