diff -up 0ad-0.0.14-alpha/build/premake/premake4.lua.orig 0ad-0.0.14-alpha/build/premake/premake4.lua
--- 0ad-0.0.14-alpha/build/premake/premake4.lua.orig	2013-08-26 23:42:56.000000000 -0300
+++ 0ad-0.0.14-alpha/build/premake/premake4.lua	2013-09-05 11:04:57.114027595 -0300
@@ -326,26 +326,6 @@ function project_set_build_flags()
 			defines { "INSTALLED_LIBDIR=" .. _OPTIONS["libdir"] }
 		end
 
-		if os.is("linux") or os.is("bsd") then
-			-- To use our local shared libraries, they need to be found in the
-			-- runtime dynamic linker path. Add their path to -rpath.
-			if _OPTIONS["libdir"] then
-				linkoptions {"-Wl,-rpath," .. _OPTIONS["libdir"] }
-			else
-				-- On FreeBSD we need to allow use of $ORIGIN
-				if os.is("bsd") then
-					linkoptions { "-Wl,-z,origin" }
-				end
-
-				-- Adding the executable path and taking care of correct escaping
-				if _ACTION == "gmake" then
-					linkoptions { "-Wl,-rpath,'$$ORIGIN'" } 
-				elseif _ACTION == "codeblocks" then
-					linkoptions { "-Wl,-R\\\\$$ORIGIN" }
-				end
-			end
-		end
-
 	end
 end
 
